java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/MethodResolutionLogic.java

```diff
@@ -208,7 +208,21 @@ private static boolean isAssignableMatchTypeParametersMatchingQName(ReferenceTyp
                     }
                 }
             } else if (expectedParam.isReferenceType()) {
-                if (!expectedParam.equals(actualParam)) {
+                if (actualParam.isTypeVariable()) {    
```

Comment:
Should we create an helper method for this?

Reply:
I'm on it :)